django/core/handlers/asgi.py

```diff
@@ -94,9 +94,14 @@ def __init__(self, scope, body_file):
             # HTTP/2 say only ASCII chars are allowed in headers, but decode
             # latin1 just in case.
             value = value.decode("latin1")
-            if corrected_name in self.META:
-                value = self.META[corrected_name] + "," + value
-            self.META[corrected_name] = value
+            if corrected_name == "HTTP_COOKIE":
+                if corrected_name in self.META:
+                    value = self.META[corrected_name] + "; " + value
```

Comment:
This isn't likely correct. The value is already terminated in `;`. 

Will be easier to see once the test is simplified.

Reply:
Likely no additional separator is needed to merge the `Cookie` headers. `parse_cookie` skips empty key/value pairs, so the surplus semicolons are harmless,  but also unnecessary.

Reply:
I think an assertion on the contents of `request.META["HTTP_COOKIE"]` is worth adding to verify this.